aptos-move/move-examples/resource_account/sources/resource_account.move

```diff
@@ -0,0 +1,129 @@
+module resource_account::resource_account {
+    use std::signer;
+    use std::string;
+
+    use aptos_framework::account;
+    use aptos_framework::coin::{Self, Coin, MintCapability, BurnCapability};
+    use aptos_framework::resource_account;
+    use aptos_framework::aptos_coin::{AptosCoin};
+
+    struct ModuleData has key {
+        resource_signer_cap: account::SignerCapability,
+    }
+
+    struct CoinCapabilities has key {
```

Comment:
Any reason to not use ManagedCoin?

Reply:
since we're returning `Coin<ChloesCoin>` in `exchange_to`, 
- if we use `managed_coin`: in `exchange_to`, we will need to do `managed_coin::mint(resource_account_signer, @resource_address, amount)` and then `coin::withdraw(resource_account_signer, amount)`
- if we store the capabilities within the `ModuleData` ourselves: in `exchange_to`, we will need to do `coin::mint<ChloesCoin>(amount, &coin_cap.mint_cap)`

I feel like approach 2 (storing the capabilities within the Module) seems simpler and will potentially save gas if this is a real-world application?